### PR TITLE
Fixing install script

### PR DESCRIPTION
**Pull request by tristanlatr**
Using `sh -c "$(curl -sSL https://raw.githubusercontent.com/lukaspustina/wpscan-analyze/master/install.sh)"` to call install script to avoid `sh` syntax error and trouble with `read` command

### CODE REVIEW

**Comment by tristanlatr** [comment]
There is some weird errors on the install script with Rasbian... 
`sh: 30: read: arg count`

**Comment by tristanlatr** [comment]
Hello,

I think the script is now stable on most unix systems.

You can try this version with `sh -c "$(curl -sSL https://raw.githubusercontent.com/lukaspustina/wpscan-analyze/41b8647ee8cccd4e24a3b3b51aa35b960098e974/install.sh)"`

**Comment by tristanlatr** [comment]
Now it's `sh -c "$(curl -sSL https://raw.githubusercontent.com/lukaspustina/wpscan-analyze/fb8bff45d9dce67829c9cccd4f56511d4fc958f8/install.sh)"`

**Comment by tristanlatr** [comment]
Hi, @lukaspustina 
The install script is currently not working.
Could you review and merge this pr ?

**Comment by lukaspustina** [comment]
I'm not sure about the contracts of the install script, but if try to re-run the script, I get this: 

```
[lukas] /tmp →  sh ./install.sh
[INFO] wpscan-analyze install script for MacOS and Linux
[INFO] wpscan-analyze in already installed: /Users/lukas/.cargo/bin/wpscan-analyze (wpscan-analyze 1.0.4)
[INFO] The script will remove this installation
[ERROR] You already have the latest wpscan-analyze version. If you're looking for the dev version please install it manually.
```

This seems odd, as the script says it's going to remove the current installation, but then bails.

File: README.md [hunk 1]
```diff
@@ -38,11 +38,11 @@ Analyzer result summary: outdated=3, vulnerabilities=1, failed=0
 11
 ```
 
-Ok => 0
```
Comment (lukaspustina) [inline README.md]: Please undo this change as it changes the contract of exit states.

File: README.md [hunk 2]
```diff
@@ -59,16 +59,28 @@ curl -s https://packagecloud.io/install/repositories/lukaspustina/opensource/scr
 sudo apt-get install wpscan-analyze
 ```
 
-### Install script
+### Install script [All Unix platforms]
 
 **Simply run:**  
 ```bash
-curl -s https://raw.githubusercontent.com/lukaspustina/wpscan-analyze/master/install.sh | sh
```
Comment (lukaspustina) [inline README.md:67]: I'd rather understand why the `curl | sh` does not work. The usual suspect is that you make assumption about what `sh` actually is. It might be bash, it might dash ... etc.

You could try to check that with your Raspi and run `shellcheck` to verify if you have any incompatibilities or ambiguous syntax -- see below.

File: install.sh [hunk 3]
```diff
@@ -12,89 +12,135 @@ binary_file="${install_to}/wpscan-analyze"
 set -e
 
 # Cd to temp and try to create/delete file
-cd /tmp/
+cd /tmp/ || ( mkdir -p tmp && cd tmp )
```
Comment (lukaspustina) [inline install.sh:21]: That's odd. Why should `cd /tmp` fail?

**Review by lukaspustina — CHANGES_REQUESTED** [review]

**Comment by lukaspustina** [comment]
I've just run shellcheck on the script and it points out some valid point: 
```
[11:15:44 vagrant@bionic64-vagrant-vm] ~  →  shellcheck -s dash install.sh

In install.sh line 22:
raw_version=`curl --silent "https://api.github.com/repos/lukaspustina/wpscan-analyze/releases/latest" | grep tag_name | sed -E 's/.*"v(.*)",/\1/'`
            ^-- SC2006: Use $(..) instead of legacy `..`.


In install.sh line 28:
    old_install=`which wpscan-analyze`
                ^-- SC2006: Use $(..) instead of legacy `..`.


In install.sh line 29:
    old_install_version=`wpscan-analyze --version`
                        ^-- SC2006: Use $(..) instead of legacy `..`.


In install.sh line 52:
    git checkout --quiet ${version}
                         ^-- SC2086: Double quote to prevent globbing and word splitting.


In install.sh line 59:
            source $HOME/.cargo/env
            ^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
            ^-- SC2169: In dash, 'source' in place of '.' is not supported.
                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In install.sh line 135:
        chmod +x ${binary_file}
                 ^-- SC2086: Double quote to prevent globbing and word splitting.
```

Can you please address these.

**Comment by lukaspustina** [comment]
Last thing: Could you reduce the questions "Change default path" and "Where" to just one question:
Set installation path [default: /usr/local/bin]

File: README.md (hunk 1 above)
Comment (tristanlatr) [reply README.md]: Hi,

So what's the contract of exit state ?

File: README.md (hunk 2 above)
Comment (tristanlatr) [reply README.md:67]: The install script use `sh` by purpose.
It also works with `bash` tho.
I'll fix the other problems and see if that's an issue

File: install.sh (hunk 3 above)
Comment (tristanlatr) [reply install.sh:21]: Yes sometimes you don't have access to `/tmp/`, it depends of your environment

File: README.md (hunk 1 above)
Comment (tristanlatr) [reply README.md]: > I'm not sure about the contracts of the install script, but if try to re-run the script, I get this:
> 
> ```
> [lukas] /tmp →  sh ./install.sh
> [INFO] wpscan-analyze install script for MacOS and Linux
> [INFO] wpscan-analyze in already installed: /Users/lukas/.cargo/bin/wpscan-analyze (wpscan-analyze 1.0.4)
> [INFO] The script will remove this installation
> [ERROR] You already have the latest wpscan-analyze version. If you're looking for the dev version please install it manually.
> ```
> 
> This seems odd, as the script says it's going to remove the current installation, but then bails.

I'll make the script allow re-install of the same version

**Comment by tristanlatr** [comment]
It steel doesn't work for me the `curl -s | bash`

Even `cat install | bash` doesn't work. It's like it says yes for me and build from source on my Mac...
Does it work for you?
It's not working on a Google cloud shell neither at https://ssh.cloud.google.com/cloudshell
But the other format with `sh -c "$()"` works.

The issue seem to be about reading input from user. It seems that doing `curl -s | bash` will open a non-interactive shell and that's an issue to read user input.

https://stackoverflow.com/questions/911168/how-to-detect-if-my-shell-script-is-running-through-a-pipe

https://www.tldp.org/LDP/abs/html/intandnonint.html

I see a solution that would be to check wether it's a interactive shell, and if not, re-launch the install script in interactive mode

**Comment by tristanlatr** [comment]
> I see a solution that would be to check wether it's a interactive shell, and if not, re-launch the install script in interactive mode

Not even sure... I've tried the following but not worked
```bash
# Test for a terminal!
#  As we recall, the -t test option checks whether the stdin, [ -t 0 ],
#+ or stdout, [ -t 1 ], in a given script is running in a terminal.
if [ -t 0 ] # stdin
then
  echo "Installing..."
else
    echo "Launching..."
    sh -c "$(curl -sSL https://raw.githubusercontent.com/tristanlatr/wpscan-analyze/patch-1/install.sh)"
    exit 0
fi

## REST of the install script
```

File: README.md (hunk 2 above)
Comment (tristanlatr) [reply README.md:67]: I'm not able to make it work with `curl | bash`.  
Please feel free to add commit to https://github.com/tristanlatr/wpscan-analyze/tree/patch-1 if tou want to edit the script

File: install.sh (hunk 3 above)
Comment (lukaspustina) [reply install.sh:21]: Out of curiosity, which system don't have /tmp or prohibit access to it?

Comment (tristanlatr) [reply install.sh:21]: Terminal emulation on android for exemple

Comment (lukaspustina) [reply install.sh:21]: LOL. Okay. Not sure we should target these systems though while complicating our main targets. What do you think?

**Comment by tristanlatr** [comment]
Hello,

About this install script, do you still want changes to the script ?
I'm not sure i can make it work with `curl | sh`... 

Right now the script on master doesn't work...

**Comment by lukaspustina** [comment]
If you don't feel like driving the script forward, I'll probably remove it, because as you say, it doesn't work right now.

**Comment by tristanlatr** [comment]
Well... 

I think this script help greatly the install process, that would be too bad to give up.
Let me try to make it work once last time with `curl | sh`.

I'll report back

**Comment by tristanlatr** [comment]
Should work now !

**Review by lukaspustina — APPROVED** [review]
Nice. Thanks for your contribution!
